reactive.js

```diff
@@ -0,0 +1,56 @@
+/* reactive engine. the code here combines functions exposed by the
+ * library into a single function to be run. i keep all code depending
+ * on Bacon here, so if somebody will want to stop using Bacon in the
+ * future, it will be easier */
+var lib = require('./lib2');
+var Bacon = require('bacon');
+
+// this is an adapter function. it gets a Pouch changes object and it
+// returns a Bacon stream
+function changesToStream(changes) {
+  return Bacon.fromBinder(function(sink) {
+    changes
+      .on('change', sink)
+      .on('error', function(text) {
+        sink(new Bacon.Error(text));
+      })
+      .on('complete', function() {
+        sink(new Backn.End());
```

Comment:
yep this will fail, thanks! i still have quite some work to do but it took some time to come up with the current structure, which allows both the library functions and the reactive logic to be fully tested. the resulting control flow in Bacon is pretty succinct, i hope that it will be also reasonably easy to follow and it will not hide any nasty error ... anyway as i said it it fully testable, so we will be able to convert any doubt in a test case. The problem is that this took more time that i wanted. The main reason is that i want the whole thing to work when the configuration document is changed on the database, which we might do from the dashboard